libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/CoreBotAdapter.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Integration.Runtime
+{
+    /// <summary>
+    /// Defines the bot runtime standard implementation of <see cref="BotFrameworkHttpAdapter"/>.
+    /// </summary>
+    internal class CoreBotAdapter : BotFrameworkHttpAdapter
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CoreBotAdapter"/> class.
+        /// </summary>
+        /// <param name="services">Services registered with the application.</param>
+        /// <param name="configuration">Application configuration.</param>
+        public CoreBotAdapter(IServiceProvider services, IConfiguration configuration)
```

Comment:
The configuration parameter shouldn't be needed, IConfiguration is already registered in services by the aspnetcore infrastructure (during ConfigureWebHostDefaults). 

We can remove this parameter and get IConfiguration from IServiceProvider.

Reply:
I don't think removing IConfiguration is the way to go. We want to use service provider less. Passing IConfiguration to allow components to configure themselves is something we've established in our samples as a positive pattern. Most adapters in our repos and the community repo were updated to receive an IConfiguration and optional Options. Our adapters in samples receive an IConfiguration as well. also this comment is the exact opposite of Johns comment about this topic.

Reply:
I don't mind the IConfiguration parameter, it's the IServiceProvider parameter that is the problem. I mean the point of the whole thing is to declaratively model your dependencies. So if you have IConfiguration as an argument you are telling the world "I depend on configuration," however having IServiceProvider as an argument is basically saying "I depend on some other stuff I'm not going to tell you about, so good luck."

There is a deeper thread that we'll have to pull on in the next milestone and that is really understanding what the dependencies actually are and what needs to be there and what doesn't. Specifically if a class here ends out having lots of arguments (aka dependencies) because we declared them correctly and clearly then that is a pointer to the factoring being wrong in the class structure. Basically you'd hope for things to have 2 or 3 dependencies ideally.

Reply:
Added the explicit real dependencies for now and next iteration in R13 we can look at these and think why our dependnecy tree is so complicated and what we can do about it.

Based on some experiments, 2 thigns will get us to a very, very concise place: Figuring out DialogManager+IBot plus introducing CloudAdapter. With those 2, these dependencies will most go away.